index.d.ts

```diff
@@ -36,13 +33,12 @@ declare class FirstChunkStream extends DuplexStream {
 
 	// unicorn.txt => unicorn rainbow
 	const stream = fs.createReadStream('unicorn.txt')
-		.pipe(new FirstChunkStream({chunkLength: 7}, (error, chunk, encoding, callback) => {
+		.pipe(new FirstChunkStream({chunkLength: 7}, async (error, chunk, encoding) => {
```

Comment:
Per my issue comment, there should be no `error` argument. That was only there because how callback-convention. It's no longer needed.

Reply:
Uh, sorry, missed that.
Are you sure about though? I don't think `error` argument is directly related to callback convention in that case. If we remove it, we will throw away a good part of library functionality(processing stream errors), and can delete half the tests and some code too.

Reply:
It should still handle errors, it's just that we don't need to pass the error to the transform callback anymore. Instead we need to `await` and try/catch the transform callback (in case an error is thrown in there).

Reply:
But that argument is here so we pass error from stream to callback, so **callback** can handle stream errors, not the opposite.

Reply:
But the intention of this package is not to be able to handle stream errors. If a user needs to handle stream errors, they can just do `stream.on('error')`.

Reply:
Ok, I removed that code and tests(one test replaced by test for errors coming from callback).